manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -303,28 +303,27 @@ func (r *FybrikApplicationReconciler) deleteExternalResources(applicationContext
 	return nil
 }
 
-// setReadModulesEndpoints populates the ReadEndpointsMap map in the status of the fybrikapplication
-func setReadModulesEndpoints(application *api.FybrikApplication, flows []api.Flow) {
-	readEndpointMap := make(map[string]taxonomy.Connection)
+// setVirtualEndpoints populates the endpoints in the status of the fybrikapplication
+func setVirtualEndpoints(application *api.FybrikApplication, flows []api.Flow) {
+	endpointMap := make(map[string]taxonomy.Connection)
 	for _, flow := range flows {
-		if flow.FlowType == taxonomy.ReadFlow {
-			for _, subflow := range flow.SubFlows {
-				if subflow.FlowType == taxonomy.ReadFlow {
-					for _, sequentialSteps := range subflow.Steps {
-						// Check the last step in the sequential flow that is for read (this will expose the reading api)
-						lastStep := sequentialSteps[len(sequentialSteps)-1]
-						if lastStep.Parameters.API != nil {
-							readEndpointMap[flow.AssetID] = lastStep.Parameters.API.Connection
-						}
-					}
-				}
+		// sanity check
+		if len(flow.SubFlows) == 0 {
+			continue
+		}
+		subflow := flow.SubFlows[len(flow.SubFlows)-1]
+		for _, sequentialSteps := range subflow.Steps {
+			// Check the last step in the sequential flow (this will expose the api)
+			lastStep := sequentialSteps[len(sequentialSteps)-1]
+			if lastStep.Parameters.API != nil {
+				endpointMap[flow.AssetID] = lastStep.Parameters.API.Connection
```

Comment:
in what case the api of the last step is not nil? is it worth to emit a log message in this case? (or an error if we always expect it to be not empty)

Reply:
Not all flows expose an api, e.g. copy or delete.